### ISSUE TITLE
Cannot connect to Postgres

### ISSUE THREAD

**Issue by MatteoJoliveau** [rickbergfalk/postgrator#45]
Hello,
I have a problem running version 3.1.0 of Postgrator with PostgreSQL.
My setup is like this: 
```javascript
this.postgrator = new Postgrator({
            migrationDirectory: dbconfig.migrations,
            schemaTable: "schema_version",
            driver: "pg",
            host: dbconfig.host,
            port: dbconfig.port,
            database: dbconfig.database,
            username: dbconfig.username,
            password: dbconfig.password,
        });
this.postgrator.runQuery("SELECT 1").then(r =>//stuff ).catch(e => {});
```
And this is the outcome:  
![image](https://user-images.githubusercontent.com/9077612/34835111-2defdc68-f6f4-11e7-82de-7fada4425807.png)
The problem is located in `lib/commonClient.js`, line 208, precisely because `commonClient.dbConnection.connect()` returns `undefined`.

**Comment by rickbergfalk** [comment]
Hi @MatteoJoliveau - may I ask what version of Node.js you are running? 

Postgrator 3 requires Nodejs 6+

**Comment by MatteoJoliveau** [comment]
Hi @rickbergfalk, I'm using Node 8.9.something (don't have access to the dev machine right now to verify the proper patch version)

**Comment by MatteoJoliveau** [comment]
Hi @rickbergfalk, I'm using Node 8.9.something (don't have access to the dev machine right now to verify the proper patch version)

**Comment by MatteoJoliveau** [comment]
It's Node v8.9.2

**Comment by rickbergfalk** [comment]
Going through the code and running tests and I think I realized what it might be...

What version of the pg module do you have installed? 

I'm wondering if the version in use has an API that is not compatible with what Postgrator expects.

**Comment by MatteoJoliveau** [comment]
I'm using pg@6.4.2. What version do you suggest to use?

**Comment by rickbergfalk** [comment]
Its tested with pg 7.x.x , but I thought it was compatible with 6.x.x as well. I can confirm that quickly though by installing that version and running tests. Hopefully I reproduce the error and can confirm thats the case :)

**Comment by rickbergfalk** [comment]
Confirmed that postgrator is not compatible with pg 6. 

If you are able to update your pg module to 7 it should work. Otherwise I'll use this issue to do some api version detection and try to add support for the previous version, or at least log a warning that pg needs a certain major version to function.

**Comment by MatteoJoliveau** [comment]
I can use PG 7 without any problem, I'm doing a major rewrite of a project so I don't have any limitations.
But yes please insert a log that postgrator needs PG 7 for other developers in the future, it's painful to understand what's going on otherwise. 
Thanks for your help!

**Comment by rickbergfalk** [comment]
No problem, and sorry about all this confusion!

**Comment by MatteoJoliveau** [comment]
I confirm that with PG 7.x.x it works, yay!

**Comment by rickbergfalk** [comment]
Great to hear! 

I've gone ahead and made some tweaks to support 6.x alongside 7.x to hopefully prevent others from experiencing weird issues. 6.x support is available in postgrator 3.2.0.

I've also updated the readme. 

I didn't look into it too much, but it doesn't look like there's a sure way to look up the version of an installed module natively in node... I'm sure there's an npm module out there to do that though. 

If I can find one I'll add a version detection. Otherwise all error messages could end up as `Some error message (or maybe unsupported driver version installed)`

**Comment by MatteoJoliveau** [comment]
You can always declare the database drivers as `optionalDependencies` or `peerDependencies` in your `package.json`, I never tested those sections but I think they may produce at least a warning log if the version doesn't match with the one installed

**Comment by rickbergfalk** [comment]
I know from other libs peer dependencies will log warnings for every peer not installed, which might be an annoyance for some. I'll have to look into optional dependencies.

I also just learned about `require.resolve(modulename)` which returns the path of the package file. I should be able to perform a version check off of that too.

**Comment by rickbergfalk** [comment]
It looks like npm tries to install optional dependencies, but doesn't stop if there is a problem.

I've published 3.3.0 to npm - now warnings will log to console if major version is outside range of support.

Thanks for reporting this issue! Hopefully there's enough here now to help others out.
